addons/website_sale_mrp/tests/test_website_sale_product_availability.py

```diff
@@ -0,0 +1,66 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.fields import Command
+from odoo.tests import HttpCase, tagged
+
+from odoo.addons.sale.tests.test_sale_product_attribute_value_config import TestSaleProductAttributeValueCommon
+
+
+@tagged('post_install', '-at_install')
+class TestWebsiteSaleMrpAvailability(HttpCase, TestSaleProductAttributeValueCommon):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        # Run the tests in another company, so the tests do not rely on the
+        # database state (eg the default company's warehouse)
+        cls.company = cls.env['res.company'].create({'name': 'Kit Company'})
+        cls.env.user.company_id = cls.company
+        cls.website = cls.env.ref('website.default_website')
+        cls.website.company_id = cls.env.company
+        # cls.website = cls.env['website'].create({'name': 'Website Kit Company'})
+        # cls.website.company_id = cls.company
+        cls.warehouse = cls.env['stock.warehouse'].search([('company_id', '=', cls.company.id)], limit=1)
+
+        # Create two storable products
+        cls.super_kit_product, cls.kit_product, cls.component_A, cls.component_B = cls.env['product.product'].create([
+            {
+                'name': f"{product_name}",
+                'allow_out_of_stock_order': False,
+                'type': 'product',
+                'website_published': True,
+                'show_availability': True,
+                'available_threshold': 100,
+            } for product_name in ("Super Kit Product", "Kit Product", "Component A", "Component B")
+        ])
+
+        cls.super_kit_bom, cls.kit_bom = cls.env['mrp.bom'].create([
```

Comment:
would be nice to also add a non store component to ensure it do not change the values